workflow/models.py

```diff
@@ -160,7 +160,19 @@ class Meta:
         verbose_name_plural = "Organizations"
         app_label = 'workflow'
 
+    def clean_fields(self, exclude=None):
+        super(Organization, self).clean_fields(exclude=exclude)
+        if self.oauth_domains:
+            for domain in self.oauth_domains:
```

Comment:
If we use `overlap` instead of `icontains`, we don't need to loop through the array of domains.
I think it's better to do it, so the database will be responsible for the operation.
Check it out!
https://docs.djangoproject.com/en/1.11/ref/contrib/postgres/fields/#overlap

Reply:
Really good advise